core/domain/takeout_service_test.py

```diff
@@ -0,0 +1,604 @@
+        general_feedback_message_data = {}
+        general_feedback_thread_data = {}
+        general_suggestion_data = {}
+        last_playthrough_data = {}
+        learner_playlist_data = None
+        incomplete_activities_data = None
+        settings_data = {
+            'email': 'user1@example.com',
+            'role': feconf.ROLE_ID_ADMIN,
+            'username': None,
+            'normalized_username': None,
+            'last_agreed_to_terms': None,
+            'last_started_state_editor_tutorial': None,
+            'last_started_state_translation_tutorial': None,
+            'last_logged_in': None,
+            'last_edited_an_exploration': None,
+            'profile_picture_data_url': None,
+            'default_dashboard': 'learner',
+            'creator_dashboard_display_pref': 'card',
+            'user_bio': None,
+            'subject_interests': [],
+            'first_contribution_msec': None,
+            'preferred_language_codes': [],
+            'preferred_site_language_code': None,
+            'preferred_audio_language_code': None
+        }
+        skill_data = {}
+        stats_data = None
+        story_progress_data = {}
+        subscriptions_data = {
+            'activity_ids': [],
+            'collection_ids': [],
+            'creator_ids': [],
+            'general_feedback_thread_ids': [],
+            'last_checked': None
+        }
+        expected_export = {
+            'user_stats_data': stats_data,
+            'user_settings_data': settings_data,
+            'user_subscriptions_data': subscriptions_data,
+            'user_skill_mastery_data': skill_data,
+            'user_contributions_data': contribution_data,
+            'exploration_user_data_data': exploration_data,
+            'completed_activities_data': completed_activities_data,
+            'incomplete_activities_data': incomplete_activities_data,
+            'exp_user_last_playthrough_data': last_playthrough_data,
+            'learner_playlist_data': learner_playlist_data,
+            'collection_progress_data': collection_progress_data,
+            'story_progress_data': story_progress_data,
+            'general_feedback_thread_data': general_feedback_thread_data,
+            'general_feedback_message_data': general_feedback_message_data,
+            'collection_rights_data': collection_rights_data,
+            'general_suggestion_data': general_suggestion_data,
+            'exploration_rights_data': exploration_rights_data,
+            'general_feedback_email_reply_to_id_data': reply_to_data
+        }
+
+        # Perform export and compare.
+        exported_data = takeout_service.export_data_for_user(self.USER_ID_1)
+        self.assertEqual(expected_export, exported_data)
+
+    def test_export_data_nontrivial(self):
+        """Nontrivial test of export_data functionality."""
+        self.set_up_non_trivial()
+        feedback_thread_model = feedback_models.GeneralFeedbackThreadModel(
```

Comment:
Just wondering, why is this not handled in the setup function?

Reply:
I need this exact model later on to create the expected data

Reply:
OK, please add a comment to explain.